src/main/java/org/terasology/adventureassets/traps/swingingblade/SwingingBladeUtilities.java

```diff
@@ -25,13 +26,14 @@ static void rotateSwingingBlade(EntityRef blade, float gameTime) {
         LocationComponent locationComponent = blade.getComponent(LocationComponent.class);
         SwingingBladeComponent swingingBladeComponent = blade.getComponent(SwingingBladeComponent.class);
         if (locationComponent != null && swingingBladeComponent.isSwinging) {
-            float t = gameTime;
             float timePeriod = swingingBladeComponent.timePeriod;
-            float pitch, A = swingingBladeComponent.amplitude, phi = swingingBladeComponent.offset;
+            float pitch;
+            float a = swingingBladeComponent.amplitude;
+            float phi = swingingBladeComponent.offset;
             float w = (float) (2 * Math.PI / timePeriod);
-            pitch = (float) (A * Math.cos(w * t + phi));
+            pitch = (float) (a * Math.cos(w * gameTime + phi));
             Quat4f rotation = locationComponent.getLocalRotation();
-            locationComponent.setLocalRotation(new Quat4f(rotation.getYaw(), pitch, rotation.getRoll()));
+            locationComponent.setLocalRotation(new Quaternionf().rotationYXZ(rotation.getYaw(), pitch, rotation.getRoll()));
```

Comment:
Same question as above

Reply:
new Qua4f(float,float,float) is the same as new Quaternionf().rotationYXZ(yaw,pitch,roll). the axises in terasology are flipped for Y and X for rotation.